dev/com.ibm.ws.artifact.zip/src/com/ibm/ws/artifact/zip/internal/JavaArchive.java

```diff
@@ -0,0 +1,575 @@
+		if ( block.length < EOCDR_MIN_SIZE ) {
+			throw new IOException("Not a valid zip file.  Less than minimum required length.");
+		}
+		
+		// We will only enter this loop if a comment is attached to the End-of-Central-Directory-Record
+		// ... or if the zip is invalid (doesn't have an End-of-Central-Directory-Record.
+		// Note that isValidEoCDRec must have EOCDR_MIN_SIZE available at the end of the block
+		// to avoid an exception because of a read past the end of the block.
+		while ( !isValidEoCDRec( block, offset, size ) ) {
+			size++;
+			if (size == 0xFFFF) {
+				int i = 0;
+			}
+			offsetFromEnd++;
+			 
+			if ( size > EOCDR_MAX_SIZE || size > _fileLength ) {
+				throw new IOException(
+						"Cannot find central directory end record in zip");
+			}
+			
+			if ( offsetFromEnd > block.length ) {
+
+				block = this.getBlockFromEndOfFile(file, size + BLOCK_SIZE - EOCDR_MIN_SIZE, BLOCK_SIZE);
+				offsetFromEnd = EOCDR_MIN_SIZE;
+			}
+			offset = block.length - offsetFromEnd;			
+		}
+		
+		// Copy only the minimum end-of-central-directory-record, but record the actual size.  
+		// We don't need to store the comment, which could be up to 64k.		
+		_eocdRecordSize = size;
+		return Arrays.copyOfRange(block, offset, offset + EOCDR_MIN_SIZE ); 
+	}
+	
+//  Alternative implementation that avoids the overlapping blocks, but always reads to the end of
+//  when retrieving a new block.  Thus the block size grows and can get quite large.	
+//
+//
+//	private byte[] getEndOfCentralDirectoryRecord(RandomAccessFile file) throws IOException {
+//		
+//		byte[] block = getBlockFromEndOfFile(file, BLOCK_SIZE);
+//		int blockLength = block.length;  // for debugging !!!!!!!!!!!!!!!!1
+//		int size = EOCDR_MIN_SIZE ;
+//		int offset = block.length - size;
+//		
+//		if ( block.length < EOCDR_MIN_SIZE ) {
+//			throw new IOException("Not a valid zip file.  Less than minimum required length.");
+//		}
+//		
+//		// We will only enter this loop if a comment is attached to the End-of-Central-Directory-Record
+//		// ... or if the zip is invalid (doesn't have an End-of-Central-Directory-Record.
+//		while ( !isValidEoCDRec( block, offset, size ) ) {
+//			size++;
+//			if ( size > block.length ) {
+//				if ( size >= EOCDR_MAX_SIZE || size > _fileLength ) {
+//					throw new IOException(
+//							"Cannot find central directory end record in zip");
+//				}
+//				block = this.getBlockFromEndOfFile(file, size + BLOCK_SIZE);
+//			}
+//			blockLength = block.length;
+//			offset = block.length - size;
+//		}
+//		
+//		// Copy only the minimum end-of-central-directory-record, but record the actual size.  
+//		// We don't need the comment, which could be up to 64k.		
+//		_eocdRecordSize = size;
+//		return Arrays.copyOfRange(block, offset, offset + EOCDR_MIN_SIZE ); 
+//	}
+	
+	/**
+	 * Get the 20-byte Zip64 Locator.  It must sit 20 bytes before the End-of-Central-Directory-Record.
+	 * @param file the zip file
+	 * @return
+	 * @throws IOException
```

Comment:
As above; @param, @return, and @throws can be omitted if they provide no actual documentation.

Reply:
done